pkg/metricutil/metricutil.go

```diff
@@ -41,34 +42,32 @@ func GetCmdLabel(request *pdpb.Request) string {
 	name := request.GetCmdType().String()
 	label, ok := cmdLabels[name]
 	if !ok {
-		label = convertName(name)
+		label = camelCaseToSnakeCase(name)
 	}
 	return label
 }
 
 func convertCmdLabels() map[string]string {
 	labels := make(map[string]string)
 	for name := range pdpb.CommandType_value {
-		labels[name] = convertName(name)
+		labels[name] = camelCaseToSnakeCase(name)
 	}
 	return labels
 }
 
-// convertName converts variable name to a linux type name.
-// Like `AbcDef -> abc_def`.
-func convertName(str string) string {
-	name := make([]byte, 0, 64)
-	for i := 0; i < len(str); i++ {
-		if str[i] >= 'A' && str[i] <= 'Z' {
-			if i > 0 {
-				name = append(name, '_')
-			}
-			name = append(name, str[i]+'a'-'A')
-		} else {
-			name = append(name, str[i])
+func camelCaseToSnakeCase(str string) string {
+	runes := []rune(str)
+	length := len(runes)
+
+	var ret []rune
+	for i := 0; i < length; i++ {
+		if i > 0 && unicode.IsUpper(runes[i]) && ((i+1 < length && unicode.IsLower(runes[i+1])) || unicode.IsLower(runes[i-1])) {
```

Comment:
In accordance with the algebraic, it is indeed the simplest expression. If format, u can split the every preconditions in one ` if line `